### DISCUSSION TITLE
Possible to expose the function that turns tailwind classes to an object?

### DISCUSSION THREAD

**Discussion by makeeno** [ben-rogerson/twin.macro#701] [Ideas]
We're trying to move away from Babel in favor of SWC and I wonder if it were possible to expose the function doing the magic behind `tw` so it can be used during runtime instead of build time.

```js
import tw from 'twin.macro'

tw`text-sm md:text-lg`

// ↓ ↓ ↓ ↓ ↓ ↓

{
  fontSize: '0.875rem',
  '@media (min-width: 768px)': {
    fontSize: '1.125rem',
  },
}
```
Or any hint how to accomplish it?

Thank you.

**Comment by ben-rogerson** [comment]
Once this version 3 business is out of the way I'm planning to start looking into this.

There's two plugin paths - esbuild and/or swc - and I'm leaning towards investigating SWC first because of it's popularity through next.js.

You're right, it'll require a separation of twin core and the babel macro, and that would allow the following combinations:
 
- twin core + twin.macro (babel setup)
- twin core + twin swc plugin (swc/next..js setup)
- twin core + twin esbuild plugin (esbuild setup)

**Reply by FranciscoKloganB** [reply]
As a user of the lib, do you think it is safe to assume that, we can use Babel setup for now and easily transition to SWC setup once it's out?

E.g.: Nothing changes except configuration and you are good to go?

I ask this because I have this project, in which I would like to keep SWC and `twin.macro`. Right now, SWC does not give me any benefits, because the project is really small. I am just wondering...

**Reply by ben-rogerson** [reply]
I'd make the upgrade process as smooth as possible.

You'll update twin.macro, then when you run your build it'll ask for another package to be installed.
Once you install that pkg, twin will continue as a babel macro just like usual.

**Comment by ben-rogerson** [comment]
For the existing SWC discussion head to https://github.com/ben-rogerson/twin.macro/discussions/516
